service/topology_coordinator.cc

```diff
@@ -752,6 +753,87 @@ class topology_coordinator : public endpoint_lifecycle_subscriber {
+                        rtlogger.error("Couldn't process global_topology_request::keyspace_rf_change, status: {}, rf per dc map: {}",
+                                        new_data.status, new_data.new_tablet_map);
+                        updates.clear(); // remove all tablets mutations ...
+                        break;           // ... and only create mutations deleting the global req
+                    }
+
+                    auto tablet_id = new_data.new_tablet_map.first_tablet();
+                    while (true) {
+                        auto& tablet_info = new_data.new_tablet_map.get_tablet_info(tablet_id);
+                        auto last_token = new_data.new_tablet_map.get_last_token(tablet_id);
+                        updates.emplace_back(replica::tablet_mutation_builder(guard.write_timestamp(), table->id())
+                                                 .set_new_replicas(last_token, tablet_info.replicas)
+                                                 .set_stage(last_token, locator::tablet_transition_stage::allow_write_both_read_old)
+                                                 .set_transition(last_token, locator::tablet_transition_kind::rf_change)
+                                                 .build());
+
+                        if (auto next_tablet = new_data.new_tablet_map.next_tablet(tablet_id); next_tablet.has_value()) {
+                            tablet_id = *next_tablet;
+                        } else {
+                            break;
+                        }
+                    }
+                }
+
+                updates.push_back(canonical_mutation(topology_mutation_builder(guard.write_timestamp())
+                                                             .set_transition_state(topology::transition_state::tablet_migration)
+                                                             .set_version(_topo_sm._topology.version + 1)
+                                                             .del_global_topology_request()
+                                                             .del_global_topology_request_id()
+                                                             .build()));
+                updates.push_back(canonical_mutation(topology_request_tracking_mutation_builder(utils::UUID{req_uuid})
+                                                             .done(error)
+                                                             .build()));
+                unsigned topo_muts_count = updates.size();
+
+                if (new_data.status == tablet_replica_calculation_status::success) {
+                    std::map<sstring, sstring> opts(new_rf_per_dc.begin(), new_rf_per_dc.end());
+                    auto ks_md = keyspace_metadata::new_keyspace(ks_name, "org.apache.cassandra.locator.NetworkTopologyStrategy",
+                                                                 opts, ks.metadata()->initial_tablets());
```

Comment:
how about the previous options?
is it guaranteed that `new_rf_per_dc` contains all the existing RF values?
as well as the `replication_factor` pseudo-rf option?

Reply:
ping

Reply:
Maybe we can somehow utilize github's resolved / unresolved comments feature and e.g. block merge until all comments are resolved?

If only there were some way for the reviewer to validate whether the comment was resolved correctly.

Reply:
> If only there were some way for the reviewer to validate whether the comment was resolved correctly.

On a related note, I don't think it should be the submitter who resolves the conversation, but the reviewer who wrote the comment, as only (s)he understands if what's been asked for has been resolved. Author of the patch would reply that a comment has been fixed and whoever wrote the comment should validate that and either continue the conversation or resolve it.
I can hardly understand why we're reversing this dependency and pretend that a submitter can safely resolve all the comments, most likely it's to create a pressure on the maintainers/reviewers and keep the ball rolling.

Reply:
> I can hardly understand why we're reversing this dependency

Resolving the comment threads is also a poor man way to keep track of which comments were addressed and which are still opened.

Personally, since we don't have a convention that the reciewer resolves discussion threads, I do that in my PRs for old comments that were addressed by some version and there were no new review comments comments about it after I annotated it that is was addressed, as a tool for myself to see what's still opened and requires my attention.

Reply:
> how about the previous options? is it guaranteed that `new_rf_per_dc` contains all the existing RF values? as well as the `replication_factor` pseudo-rf option?

The `replication_factor` tag is removed, and yes, I'm only passing new RF-per-DC values.